text/0019-clipboard-rearchitecture.md

```diff
@@ -0,0 +1,299 @@
+ 
+#### Preferred Approach/Solution
+
+This design recommends alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t considered by the specification.  Given this approach, we will preserve some of the existing APIs, remove some of the existing APIs, and, modify the rest of the APIs to be in compliance with the W3C spec as much as possible.
+
+**APIs to Preserve**
+* `clipboard.clear([type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readText([type])`
+  * Already implemented per specification.
+* `clipboard.writeText(text[, type])`
+  * Already implemented per specification.
+* `clipboard.has(format[, type])` 
+  * Convenience method to quickly check if a particular format is available.
+
+**APIs to Remove**
+* `clipboard.availableFormats([type])`
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.readBookmark()`
+  * Superseded by `clipboard.read` with custom `electron/bookmark` MIME type
+* `clipboard.writeBookmark(title, url[, type])`
+  * Superseded by `clipboard.write` with custom `electron/bookmark` MIME type
+* `clipboard.readBuffer(format)`
+  * Superseded by `clipboard.read`[Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.writeBuffer(format, buffer[, type])`
+  * Superseded by `clipboard.write` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write)
+* `clipboard.readFindText()` _macOS_
+  * Superseded by `clipboard.read` with custom `electron/findtext` MIME type
+* `clipboard.writeFindText(text)`
+  * Superseded by `clipboard.write` with custom `electron/findtext` MIME type
+* `clipboard.readHTML()`
+  * Superseded by `clipboard.read` with `text/html` MIME type
+* `clipboard.writeHTML(markup[, type])`
+  * Superseded by `clipboard.write` with `text/html` MIME type
+* `clipboard.readImage([type])`
+  * Superseded by `clipboard.read()` with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.writeImage(image[, type])`
+  * Superseded by `clipboard.write`  with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.readRTF([type])`
+  * Superseded by `clipboard.read` with `application/rtf` MIME type
+* `clipboard.writeRTF(text[, type])`
+  * Superseded by `clipboard.write` with `application/rtf` MIME type
+
+**APIs to Modify**
+* `clipboard.read([clipboardType])`
+  * `clipboardType` string (optional) - Can be `selection` or `clipboard`; default is `clipboard`. `selection` is only available on Linux.
+  * This API will be modified to bring into spec compliance with the [Web API clipboard.read](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+  * Returns a Promise that resolves with an array of [ClipboardItem](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem) objects containing the 
```

Comment:
From the [ClipboardItem docs](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem), it seems like a single `ClipboardItem` has a `types` array which lets it contain multiple clipboard types. Would we need to return an array of `ClipboardItem`, or would one `ClipboardItem` be sufficient?